### PR TITLE
[SD] Principal stress output

### PR DESCRIPTION
**Pull request by nagelt**
Adds principal stress and principal stress direction output to the SmallDeformation process.
The implementation is in principle the same as the one used in the HydroMechanics process. The hollow sphere test is likewise the same.

### CODE REVIEW

**Comment by nagelt** [comment]
@FZill Maybe you can have a look, since you did the HM version.

File: ProcessLib/SmallDeformation/SmallDeformationFEM.h [hunk 1]
```diff
@@ -493,6 +493,50 @@ class SmallDeformationLocalAssembler
...
+        x_position.setElementID(elem_id);
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        constexpr int symmetric_tensor_size =
+            MathLib::KelvinVector::KelvinVectorDimensions<
+                DisplacementDim>::value;
+
+        auto sigma_sum = MathLib::KelvinVector::tensorToKelvin<DisplacementDim>(
+            Eigen::Matrix<double, 3, 3>::Zero());
```
Comment (TomFischer) [inline ProcessLib/SmallDeformation/SmallDeformationFEM.h:510]: Is there a relation between the DisplacementDim the fixed size 3 of the matrix?

Comment (nagelt) [reply ProcessLib/SmallDeformation/SmallDeformationFEM.h:510]: not in this case. We're always dealing with 3 normal stresses, from which we calculate the principal values.

Comment (TomFischer) [reply ProcessLib/SmallDeformation/SmallDeformationFEM.h:510]: Thanks for answer.

**Comment by FZill** [comment]
Looks good to me. Regarding the output to be optional - If we do that, we should do it in HM too.

**Comment by nagelt** [comment]
> Looks good to me. Regarding the output to be optional - If we do that, we should do it in HM too.

I think so, too. Maybe this could be a follow-up PR that makes this feature in both processes optional.

**Comment by wenqing** [comment]
> > Looks good to me. Regarding the output to be optional - If we do that, we should do it in HM too.
> 
> I think so, too. Maybe this could be a follow-up PR that makes this feature in both processes optional.

That is good.

**Pull request merged**